carbon-design-system/carbon-addons-iot-react
# fix(table): prevent column widths below min width

Closes #2474

**Summary**
Fix to prevent the table from rendering column widths below the min width on the initial rendering

**Change List (commits, features, bugs, etc)**
- Removed old (CSS) style prop containing widths that was not being used
- Added new check on initial rendering that changes widths below the minimum to the default width
- Added unit test

Instead of affected widths being changed to the actual min-width (62px) I used the default width of 150px instead after discussion with @JordanWSmith15. We reasoned that the users would rarely want the 62px width since it doesn't show much of the column header. The 62px min width still stays as the lower boundary for when the user resizes the columns.

**Acceptance Test (how to verify the PR)**
Go to https://deploy-preview-2669--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-table--with-options-to-explore-column-settings use the default knob coniguration

**- Test initial column widths**
1. Use config: Initial column width: undefined, table container width: 300px, hasResize:true, wrapCellText: always, useAutoTableLayoutForResize:false, preserveColumnWidths:false
2. Make sure the columns get the default width of 150px. Before the fix the widths would have been below min width.
3. Change "table container width" to 800px. Verify that the columns get a width of 100px. This is the unmodified width set by the browser which we don't modify since it is above the min width.
4. Change "table container width" to "auto" (for a wide browser window) and make sure the columns widths are distributed evenly  

**- Test columns widths on window resize since it uses the same code**
1. Use config: Initial column width: undefined, table container width: auto, hasResize:true, useAutoTableLayoutForResize:true, preserveColumnWidths:false
2. Resize browser window and make sure column widths still adjust to the text of the columns headers and that the min-width of 62 pixels is respected.

**Regression Test (how to make sure this PR doesn't break old functionality)**
Check that resize and column add/remove/toggle works as expected



**Things to look for during review**
